heron/tmaster/src/cpp/manager/tmaster.cpp

```diff
@@ -106,6 +104,12 @@ TMaster::TMaster(const std::string& _zk_hostport, const std::string& _topology_n
   do_reassign_ = false;
 
   master_establish_attempts_ = 0;
+
+  // start servers to get the actual ports before `tmaster_location_` related operation
```

Comment:
this breaks one fundamental invariant in tmaster design. No operations before verifying that we are the master.